package messages;

message WorldModel
{
    optional int64 timestamp = 1;

    optional float my_x = 2;
    optional float my_y = 3;
    optional float my_h = 4;

    optional float my_uncert = 5;

    optional bool  ball_on = 6;

    optional float ball_dist = 7;
    optional float ball_bearing = 8;

    optional float ball_dist_uncert = 9;
    optional float ball_bearing_uncert = 10;

    optional float chase_time = 11;
    optional float defender_time = 12;
    optional float offender_time = 13;
    optional float middie_time = 14;

    optional int32 role = 15;
    optional bool  in_kicking_state = 16;

    optional bool  active = 17;

    optional float shared_ball_x = 18;
    optional float shared_ball_y = 19;
}